Next VisiData Release

anjakefala opened the pull request.

abitrolly committed:
dev/mkman.sh: clean and create /tmp dir automatically

abitrolly committed:
Catch man page build errors and add the check to CI

abitrolly committed:
dev/mkman.sh requires `aha` to be installed

anjakefala committed:
Merge branch 'develop' into mandown

anjakefala committed:
[dev] bump to v2.11dev

anjakefala committed:
[open-cell-file] warning when file or url in cell does not exist

Closes #1540

anjakefala committed:
[dev] remove --universal; we are not building for python 2

saulpw committed:
[graph-] fail if no numeric xcols given

saulpw committed:
[unzip-http-] fix extract

saulpw committed:
[deprected] refactor into deprecated_warn

saulpw committed:
[api] Sheet.StaticColumn renamed to Sheet.freeze_col

saulpw committed:
[columns] SettableColumn now copied with sheet

saulpw committed:
[zip] multisave

edupont committed:
Typo (#1573)

* Typo

* Update customize.md

Co-authored-by: Saul Pwanson <code@saul.pw>

Freed-Wu committed:
Fix #1583

saulpw committed:
[describe- expr-] use vd.getGlobals() instead of globals() #1574

anjakefala committed:
[tests] add Python 3.11 to tests

Closes #1585

anjakefala committed:
[docs] update test badge to github actions and add discord badge

Closes #1586

anjakefala committed:
[dev] ensure setuptools files have appropriate perms #1591

saulpw committed:
[dirsheet] add open-dir-parent (bound to `)

saulpw committed:
[docs] update pipes doc

saulpw committed:
[docs] update viewtsv  #1610

saulpw committed:
[geojson-] move columns to class level

saulpw committed:
[sqlite] allow any "INT" type abbrev

saulpw committed:
[freqtbl] dive-rows becomes dive-selected; to openRows

saulpw committed:
[aggregators] add 95 and 99 pctile

dbaynard committed:
Store nulls as empty strings in save_xlsx

Before this change, nulls in the input would result in `None` in the
saved xlsx files.

cool-RR committed:
Fix exception causes in utils.py

cool-RR committed:
Fix exception causes, remove invalid exception

saulpw committed:
[docs-] fix misspellings #1640

Co-author: Kian-Meng Ang <kianmeng@cpan.org>

saulpw committed:
[freqtbl] add HistogramColumn to allow overrides #1621

To replace the histogram function, use:

    @HistogramColumn.api
    def calcValue(col:Column, row) -> str:
        ...

saulpw committed:
[api] easier external numeric types #1394

- add vd.numericTypes mutable list
- add @vd.numericType() decorator for external types
- refactor floatsi into visidata.type_floatsi
- refactor date into visidata.type_date
- refactor currency into visidata.type_currency

Now you can make e.g. `log10` a valid numeric type for a column:

    @vd.numericType('/')
    def log10(*args):
        if not args:
            return 0
        return math.log10(*args) if args[0] > 0 else -10

    Sheet.addCommand('', 'type-log10', 'cursorCol.type = log10')

which might make it nicer to work with.

saulpw committed:
[api] addGlobals can also take kwargs

saulpw committed:
[path-] fix missing import

saulpw committed:
[customdate-] add to numericTypes; set fmtstr on existing cols #1613

anjakefala committed:
[build] drop testing support for Python 3.6

Related to: https://github.com/actions/setup-python/issues/543

anjakefala committed:
[freeze] add missing import

anjakefala committed:
[tests] fix typo in column name

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/saulpw/visidata?pullRequest=1646) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/saulpw/visidata?pullRequest=1646) before we can accept your contribution.<br/>**7** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: abitrolly<br/>:white_check_mark: anjakefala<br/>:white_check_mark: Freed-Wu<br/>:white_check_mark: saulpw<br/>:white_check_mark: cool-RR<br/>:white_check_mark: midichef<br/>:white_check_mark: dbaynard<br/>:x: edupont<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/saulpw/visidata?pullRequest=1646) it.</sub>

saulpw committed:
[api] add vd.launchExternalEditorPath

saulpw committed:
[sysedit-selected] edit rows in $EDITOR #1596

- edit rows
   - save to temp file in given format (default tsv)
   - edit in $EDITOR
   - load values in temp sheet
   - setValuesTyped on existing sheet
- only handles cell modifications, not added/deleted rows/cols
- bound to gCtrl+O for now

saulpw committed:
[formats] rename "VisiData loader" column to "loader"

anjakefala committed:
[dev] update CHANGELOG

anjakefala committed:
[dev] update CHANGELOG

anjakefala committed:
[dev] drop support for v3.6

saulpw committed:
[join append] simple append for sheets of same type #1598

Using `append` join with sheets of the same type copies the first sheet
and creates a new rowset with a concatenation the original rows, instead
of creating a special ConcatSheet.

anjakefala committed:
Merge branch 'develop' into mandown

anjakefala committed:
Merge pull request #1520 from abitrolly/mandown

CI/CD check for man pages build

saulpw committed:
[html] make relative links absolute #1599

saulpw committed:
[macros-] override CLI parsing options for MacrosSheet #1607

anjakefala committed:
[sheets] sort all sheets on global Sheets Sheet

Global Sheets Sheet only has precious sheets, so it is not
self-referential.

This means the standard sheet sort applies to it, it does not need
to skip the reference to itself like the standard Sheets Sheet does.

Closes #1620

saulpw committed:
[join-] add new "concat" jointype to act like "append" but keep first sheet type and columns #1598

- changed the descriptions of jointypes to be more regular
- new "concat" jointype is misnamed; probably "concat" and "append" should swap semantics

saulpw committed:
[LazyChainMap-] add get() method #1317

saulpw committed:
[LazyChainMap-] remove / from get() method #1317

saulpw committed:
[macros-] remove extra debug status

saulpw committed:
[cliptext] show disp_oddspace for surrogate escapes #1544

saulpw committed:
[hdf5-] load array of scalars #1602

- handled differently than array of compound datatype

midichef committed:
Fix typo

saulpw committed:
[dirsheet-] use internal PathLike.name instead of reconstructing #1571

saulpw committed:
[columns-] SettableColumn should not be deferred #1568

- add Column.defer attribute
   - non-deferred columns are set immediately, even on deferred sheets
- columns added during load share sheet.defer; user-added columns are not deferred
- goes with f047e4c54

saulpw committed:
[unzip-http-] handle files smaller than 64k #1567

saulpw committed:
[sqlite] add passthrough options for connect() #1622

Query parameters from https://www.sqlite.org/uri.html are like `options.sqlite_param_mode`, for the `mode` parameter.

Connection kwargs from https://docs.python.org/3/library/sqlite3.html#sqlite3.connect are like `options.sqlite_connect_isolation_level` for the `isolation_level` argument.

For example, `options.sqlite_param_immutable=1` to indicate the medium is immutable, so it won't create .wal files even in readonly mode.

anjakefala committed:
[open_txt] check if file exists before probing for tsv

anjakefala committed:
[tests] since #1611 (62caa353a) tests do not need a .visidatarc

midichef committed:
Speed up fill for sheets with many empty cells (#1657)

anjakefala committed:
[tests] add smoketests for replaying macros #1652

anjakefala committed:
[tests] fix github actions

anjakefala committed:
[tests] a .vd* must follow -p

anjakefala committed:
[tests] update golden

anjakefala committed:
[macros] try-again if keystroke already in use by a macro

Closes #1658

saulpw committed:
[keys-] add kDN/kUP to translation table #1336

anjakefala committed:
[macros] do not include nonLogged commands in macro #1569

anjakefala committed:
[macros] add reload for MacroSheet #1569

saulpw committed:
[mouse-] look for registered onMouse on all screens #1625

saulpw committed:
[status-] open sheets-stack for left click on sheet name #1625

anjakefala committed:
[macros] specificy macro is set to keybinding #1569
